warehouse/query-core/src/main/java/datawave/core/iterators/TermFrequencyOffsetIterator.java

```diff
@@ -0,0 +1,163 @@
+    
+    private SortedKeyValueIterator<Key,Value> source;
+    
+    public TermFrequencyOffsetIterator(Set<Text> searchSpace) {
+        this(new TreeSet<>(searchSpace));
+    }
+    
+    public TermFrequencyOffsetIterator(TreeSet<Text> searchSpace) {
+        this.searchSpace = searchSpace;
+    }
+    
+    public TermFrequencyOffsetIterator(TermFrequencyOffsetIterator other, IteratorEnvironment env) {
+        this.source = other.getSource().deepCopy(env);
+        this.searchSpace = other.searchSpace;
+    }
+    
+    @Override
+    public void init(SortedKeyValueIterator<Key,Value> source, Map<String,String> options, IteratorEnvironment env) throws IOException {
+        this.source = source;
+    }
+    
+    @Override
+    public void next() throws IOException {
+        this.tk = null;
+        this.tv = null;
+        if (source.hasTop()) {
+            source.next();
+            findTop();
+        }
+    }
+    
+    @Override
+    public void seek(Range range, Collection<ByteSequence> cfs, boolean inclusive) throws IOException {
+        this.initialRange = new Range(range);
+        this.tk = null;
+        this.tv = null;
+        
+        source.seek(range, seekCFs, true);
+        findTop();
+    }
+    
+    public void findTop() throws IOException {
+        this.tk = null;
+        this.tv = null;
+        Key key;
+        while (true) {
+            // If the source iterator ran out keys we're done.
+            if (!source.hasTop()) {
+                break;
+            }
+            
+            key = source.getTopKey();
+            
+            // If the source iterator has keys outside of our range, we're done.
+            if (!initialRange.contains(key)) {
+                break;
+            }
+            
+            if (searchSpace.contains(key.getColumnQualifier())) {
+                this.tk = key;
+                this.tv = source.getTopValue();
+                break;
+            } else {
+                // Always seek, for now.
+                if (!seekToNextHit(key)) {
+                    break;
+                }
+            }
+        }
+    }
+    
+    private boolean seekToNextHit(Key key) throws IOException {
+        Range seekRange = getNextSeekRange(key);
+        if (seekRange == null)
+            return false;
+        source.seek(seekRange, seekCFs, true);
+        return true;
+    }
+    
+    private Range getNextSeekRange(Key key) {
+        Text nextCQ = searchSpace.higher(key.getColumnQualifier());
+        if (nextCQ == null)
+            return null;
+        
+        // Update the start key's column qualifier
+        Key start = initialRange.getStartKey();
+        start = new Key(start.getRow(), start.getColumnFamily(), nextCQ);
+        
+        // Update the initial range with the new start key
+        return new Range(start, true, initialRange.getEndKey(), initialRange.isEndKeyInclusive());
+    }
```

Comment:
Consider minimizing the diffs between this class and `TermFrequencyIterator` so that it is easy to compare implementations. For example consider moving this entire section to the end of the class.

Reply:
It irks me that the `TermFrequencyIterator` has getters and setters before the main functions like `next`, `seek`, and `getNextSeekRange`. Going to ask forgiveness in advance for the nasty diff between the two classes and leave it as-is.